pkg/cloudprovider/vsphere/loadbalancer/config/types_yaml.go

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2020 The Kubernetes Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package config
+
+/*
+	TODO:
+	When the INI based cloud-config is deprecated, this file should be renamed
+	from types_yaml.go to types.go and the structs within this file should be named:
+
+	LBConfigYAML -> LBConfig
+	LoadBalancerClassConfigYAML -> LoadBalancerClassConfig
+	LoadBalancerClassConfigYAML -> LoadBalancerClassConfig
+	NsxtConfigYAML -> NsxtConfig
+*/
+
+// LBConfigYAML  is used to read and store information from the cloud configuration file
+type LBConfigYAML struct {
+	LoadBalancer      LoadBalancerConfigYAML                  `yaml:"loadbalancer"`
+	LoadBalancerClass map[string]*LoadBalancerClassConfigYAML `yaml:"loadbalancerclass"`
+	NSXT              NsxtConfigYAML                          `yaml:"nsxt"`
+}
+
+// LoadBalancerConfigYAML contains the configuration for the load balancer itself
+type LoadBalancerConfigYAML struct {
+	Size             string            `yaml:"size"`
+	LBServiceID      string            `yaml:"lb-service-id"`
+	Tier1GatewayPath string            `yaml:"tier1-gateway-path"`
+	AdditionalTags   map[string]string `yaml:"tags"`
+
+	// this struct use to inherit from LoadBalancerClassConfigYAML, but the YAML parser
+	// wasnt able to indirectly parse inherited fields
+	IPPoolName        string `yaml:"ip-pool-name"`
```

Comment:
`ip-pool-name` makes sense for INI but I think best practice for yaml is to use camel case?

Reply:
Oops. Good catch... it's what happens when you copy and paste using INI as the base for the go file.